pandas-stubs/core/frame.pyi

```diff
@@ -223,9 +220,11 @@ class DataFrame:
     def query(self, expr: _str) -> DataFrame: ...
     def reindex(self, index: Index) -> DataFrame: ...
     @overload
-    def rename(self, mapper: Callable, axis: _AxisType = ...) -> DataFrame: ...
+    def rename(self, mapper: Union[Dict, Callable], axis: _AxisType = ...) -> DataFrame: ...
```

Comment:
Can we say anything about the key and value type of the `Dict`? Maybe
```python
Dict[Union[str, int], Union[str, int]]
```

(This is actually related to a larger problem: what type can column identifier have? Is it only string and int?)

Reply:
We could restrict it to str and int as the most common cases.
I suppose the column identifier can have any hashable type (for ex. tuple is allowed, while list is not), so I we could also restrict it to
```
Dict[Hashable, Hashable]
```

Reply:
`Hashable` sounds great!

Reply:
done :)